ckanext/dcat/profiles.py

```diff
@@ -1306,9 +1328,16 @@ def graph_from_dataset(self, dataset_dict, dataset_ref):
             # If no name but an URI is available, the name literal remains empty to
             # avoid mixing organization and dataset values.
             if not publisher_name and not publisher_uri and dataset_dict.get('organization'):
-                publisher_name = dataset_dict['organization']['title']
-
-            g.add((publisher_details, FOAF.name, Literal(publisher_name)))
+                try:
+                    org_dict = toolkit.get_action(u'organization_show')({u'user': toolkit.g.user},
```

Comment:
It's a bit annoying that we need to have this call as it slows things down. We need to implement some sort of caching because the serialization generally happens on large numbers of datasets and we can't keep calling the same action over and over. Something like a class level `self._org_cache` that gets reset when the class is initialized.

```
org_id = dataset_dict["organization"]["id"]
if org_id in self._org_cache:
    org_dict = self._org_cache[org_id]
else: 
    org_dict = toolkit.get_action("organization_show")({"ignore_auth": True}, {"id": org_id})
    self._org_cache[org_id] = org_dict
```
Also note we need to use `{"ignore_auth": True}` otherwise this won't work outside a web request (eg [tests](https://github.com/ckan/ckanext-dcat/actions/runs/5321585352/jobs/9636871682))

You'll also need to tweak this tests to get it passing:

```diff
diff --git a/ckanext/dcat/tests/test_euro_dcatap_profile_serialize.py b/ckanext/dcat/tests/test_euro_dcatap_profile_serialize.py
index e0f7d3c..d9bbe47 100644
--- a/ckanext/dcat/tests/test_euro_dcatap_profile_serialize.py
+++ b/ckanext/dcat/tests/test_euro_dcatap_profile_serialize.py
@@ -398,13 +398,16 @@ class TestEuroDCATAPProfileSerializeDataset(BaseSerializeTest):
         assert self._triple(g, publisher, DCT.type, URIRef(extras['publisher_type']))
 
     def test_publisher_org(self):
+
+        org = factories.Organization()
+
         dataset = {
             'id': '4b6fe9ca-dc77-4cec-92a4-55c6624a5bd6',
             'name': 'test-dataset',
             'organization': {
-                'id': '',
-                'name': 'publisher1',
-                'title': 'Example Publisher from Org',
+                'id': org["id"],
+                'name': org["name"],
+                'title': org["title"]
             }
         }
```

Reply:
@amercader SURPRISE! Back to this one! Haha, had just a little free time before our upgrade to Python3 and CKAN 2.10

Wow, this was from June eh? WELL

There is now a class variable to cache the org dicts. I renamed `all_translated` to just `translated`. AND fixed that test above